bot/converters.py

```diff
@@ -495,22 +495,55 @@ async def convert(self, ctx: Context, argument: str) -> int:
         return duration
 
 
-class UserMentionOrID(UserConverter):
+def _is_an_unambiguous_user_argument(argument: str) -> bool:
+    """Check if the provided argument is a user mention, user id, or username."""
+    has_id_or_mention = bool(IDConverter()._get_id_match(argument) or RE_USER_MENTION.match(argument))
+
+    if not has_id_or_mention:
+        if argument[0] == '@':
+            argument = argument[1:]
+
+        # Check to see if the author passed a username (a discriminator exists)
+        if len(argument) > 5 and argument[-5] == '#':
+            return True
+
+    return has_id_or_mention
+
+
+AMBIGUOUS_ARGUMENT_MSG = ("`{argument}` is not a User mention, a User ID or a Username in the format"
+                          " `name#discriminator`.")
+
+
+class UnambiguousUser(UserConverter):
     """
-    Converts to a `discord.User`, but only if a mention or userID is provided.
+    Converts to a `discord.User`, but only if a mention, userID or a username is provided.
 
-    Unlike the default `UserConverter`, it doesn't allow conversion from a name or name#descrim.
+    Unlike the default `UserConverter`, it doesn't allow conversion from a name.
     This is useful in cases where that lookup strategy would lead to ambiguity.
     """
 
     async def convert(self, ctx: Context, argument: str) -> discord.User:
-        """Convert the `arg` to a `discord.User`."""
-        match = self._get_id_match(argument) or RE_USER_MENTION.match(argument)
+        """Convert the `argument` to a `discord.User`."""
+        if _is_an_unambiguous_user_argument(argument):
+            return await super().convert(ctx, argument)
+        else:
+            raise BadArgument(AMBIGUOUS_ARGUMENT_MSG.format(argument=argument))
+
+
+class UnambiguousMember(MemberConverter):
+    """
+    Converts to a `discord.Member`, but only if a mention, userID or a username is provided.
+
+    Unlike the default `MemberConverter`, it doesn't allow conversion from a name or nickname.
+    This is useful in cases where that lookup strategy would lead to ambiguity.
```

Comment:
```suggestion
    This is useful in cases where that lookup strategy would lead to too much ambiguity.
```

Reply:
> Did you also adjust tests?

By adjusting tests I'm assuming you mean adding new tests for these converters since there aren't any? I didn't do so as UserMentionOrID didn't have any tests - on what basis do we decide to add new tests for something?

Reply:
Thanks for the suggestions, I've implemented them